docs/en/modules/gpio.md

```diff
@@ -118,20 +120,20 @@ This function is not available if GPIO_INTERRUPT_ENABLE was undefined at compile
 #### Example
 
 ```lua
--- use pin 1 as the input pulse width counter
-pin = 1
-pulse1 = 0
-du = 0
-gpio.mode(pin,gpio.INT)
-function pin1cb(level)
-  du = tmr.now() - pulse1
-  print(du)
-  pulse1 = tmr.now()
-  if level == gpio.HIGH then gpio.trig(pin, "down") else gpio.trig(pin, "up") end
+do
+  -- use pin 1 as the input pulse width counter
+  local pin, pulse1, du, now, trig = 1, 0, 0, tmr.now, gpio.trig
+  gpio.mode(pin,gpio.INT)
+  local function pin1cb(level)
+    local pulse2 = now()
```

Comment:
Yup, I was just using the Lua example to show a few techniques that speed execution.this removed a couple of table look ups.